subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ctr_drbg.c

```diff
@@ -259,3 +259,22 @@ psa_status_t cracen_free_random(cracen_prng_context_t *context)
 	(void)context;
 	return PSA_SUCCESS;
 }
+
+
+psa_status_t cracen_get_trng(uint8_t *output, size_t output_size)
+{
+	int sx_err;
+	if (output == NULL || output_size == 0) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	/* Both the DRBG and the TRNG use the same mutex for synchronization,
+	 * there is no need to create another mutex for this so the existing one
+	 * is being used.
+	 */
+	nrf_security_mutex_lock(cracen_prng_context_mutex);
+	sx_err = trng_get_entropy((char *)output, output_size);
```

Comment:
I might be wrong, but should this not loop until you get enough output? Otherwise you will be limited to the maximum you can get from a single call to `trng_get_entropy`  (64 bit?)  which you don't seem to check for anywhere.

Reply:
You are right, I updated the PR to use max 32 bytes at a time.